### ISSUE TITLE
Snowflake: Parsing Error for CREATE TABLE with TAG

### ISSUE THREAD

**Issue by gopisaba** [sqlfluff/sqlfluff#3786]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Trying to integrate sqlfluff with our snowflake repository. We are using Liquibase not DBT. When running lint/fix aganist one of our sql file, it is failing with parsing error. I think sqlfluff doesn't like `.` in the tag key name.

```
L:  26 | P:  25 |  PRS | Line 26, Position 25: Found unparsable section: ' WITH
                       | TAG (\n account_objects.tag...'
```

### Expected Behaviour

Highlight Lint errors

### Observed Behaviour

Failing with Parsing errors

### How to reproduce

```sql
CREATE OR REPLACE TABLE SCHEMA1.TABLE1
(
    "COL1" varchar(128) NOT NULL,
    "COL2" varchar(128) NOT NULL
) CHANGE_TRACKING = TRUE WITH TAG (
    account_objects.tags.IRM = '{"IRM":[{"Primary":"ABC123"}]}'
);
```

### Dialect

snowfalke

### Version

1.2.1

### Configuration

[sqlfluff]
dialect = snowflake
templater = raw

[sqlfluff:rules:L057]
additional_allowed_characters = $

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
